File: sorting/SelectionSort/python/SelectionSort.py
from __future__ import annotations


def selection_sort(values: list[int]) -> None:
    """Sort values in ascending order in place using classic Selection Sort."""
    n = len(values)
    for i in range(n):
        min_index = i
        for j in range(i + 1, n):
            if values[j] < values[min_index]:
                min_index = j
        if min_index != i:
            values[i], values[min_index] = values[min_index], values[i]

File: sorting/SelectionSort/java/SelectionSort.java
public final class SelectionSort {
    private SelectionSort() {}

    public static void selectionSort(int[] values) {
        for (int i = 0; i < values.length; i++) {
            int minIndex = i;
            for (int j = i + 1; j < values.length; j++) {
                if (values[j] < values[minIndex]) {
                    minIndex = j;
                }
            }

            if (minIndex != i) {
                int temporary = values[i];
                values[i] = values[minIndex];
                values[minIndex] = temporary;
            }
        }
    }
}
